synapse/http/proxyagent.py

```diff
@@ -220,7 +221,7 @@ def request(
                 self._reactor, parsed_uri.host, parsed_uri.port, **self._endpoint_kwargs
             )
 
-        logger.debug("Requesting %s via %s", uri, endpoint)
+        logger.debug("Requesting %s via %s", redact_uri(str(uri)), endpoint)
```

Comment:
Is this good enough to convert a `bytes` to `string`?

Seems to work at least: `type(str(b'asdf'))`

Reply:
You probably want to decode it in such a way it won't fail... 

```suggestion
        logger.debug("Requesting %s via %s", redact_uri(uri.decode("ascii", errors="replace")), endpoint)
```

Reply:
Fancy! Thanks for the tip 🙂

> The errors argument specifies the response when the input string can’t be converted according to the encoding’s rules. Legal values for this argument are `'strict'` (raise a [UnicodeDecodeError](https://docs.python.org/3/library/exceptions.html#UnicodeDecodeError) exception), `'replace'` (use `U+FFFD`, `REPLACEMENT CHARACTER`), `'ignore'` (just leave the character out of the Unicode result), or `'backslashreplace'` (inserts a `\xNN` escape sequence). The following examples show the differences:
>
> https://docs.python.org/3/howto/unicode.html#the-string-type